modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/delegate/SourceModelLoaderDelegate.java

```diff
@@ -121,4 +187,72 @@ private void declareSourceCallbackParameters(Optional<SourceCallbackModelParser>
     parser.ifPresent(callback -> loader.getParameterModelsLoaderDelegate().declare(declarer.get(),
                                                                                    callback.getParameterGroupModelParsers()));
   }
+
+  private void declareMetadataKeyIdModelProperty(SourceDeclarer sourceDeclarer,
+                                                 Optional<OutputResolverModelParser> outputResolverModelParser,
+                                                 Optional<KeyIdResolverModelParser> keyIdResolverModelParser) {
+    if (keyIdResolverModelParser.isPresent()) {
+      String parameterName = keyIdResolverModelParser.get().getParameterName();
+      MetadataType metadataType = keyIdResolverModelParser.get().getMetadataType();
+      String categoryName = JavaModelLoaderUtils.getCategoryName(keyIdResolverModelParser.orElse(null), emptyList(),
```

Comment:
If the method used is exclusive to java, they it shouldn't be used here. If it is not exclusive to java, move it to a more general utils class.